ast.ts

```diff
@@ -42,6 +43,7 @@ export type Expr<A> =
   | {  a?: A, tag: "index", obj: Expr<A>, index: Expr<A> }
   | {  a?: A, tag: "method-call", obj: Expr<A>, method: string, arguments: Array<Expr<A>> }
   | {  a?: A, tag: "construct", name: string }
+  | {  a?: A, tag: "tuple", expr: Expr<A>[]}
```

Comment:
I see you went ahead with a tuple implementation. I am not sure if this is part of your previous milestone goals. I would suggest interacting with the sets/tuples team to avoid any conflicts. I am pretty sure there will be a merge conflict.

If you need an iterator to work with, working with a range class would be the best option. Any reason why you didn't go that route?

Reply:
We did not support Range in this milestone in order to re-use the For loops implementation of Range class. We intend to support that as part of the final Merge changes on Monday, by pulling in their changes. For Milestone 1, we have implemented all the basic cases (including ignore and starred) outside of Iterators and Data Structures.

We have thus removed instances of Range and Tuple (supported by other teams) from our commits to avoid conflicts.

Reply:
We have now added support for Range in this pull request.